packages/core/src/State.ts

```diff
@@ -137,70 +145,57 @@ export class State<
   public static create<
     TC extends MachineContext,
     TE extends EventObject = EventObject
-  >(config: StateConfig<TC, TE>): State<TC, TE, any> {
-    return new State(config);
+  >(config: StateConfig<TC, TE>, machine: AnyStateMachine): State<TC, TE, any> {
+    return new State(config, machine);
   }
   /**
    * Creates a new `State` instance for the given `stateValue` and `context` with no actions (side-effects).
    * @param stateValue
    * @param context
    */
-  public static inert<TState extends State<any, any, any>>(
-    state: TState
-  ): TState;
-  public static inert<
-    TContext extends MachineContext,
-    TEvent extends EventObject = EventObject
-  >(stateValue: StateValue, context: TContext): State<TContext, TEvent>;
-  public static inert(
-    stateValue: State<any, any, any> | StateValue,
-    context?: MachineContext
-  ): State<any, any, any> {
-    if (stateValue instanceof State) {
-      if (!stateValue.actions.length) {
-        return stateValue;
-      }
-      const _event = initEvent as SCXML.Event<any>;
+  public static inert(state: AnyState): State<any, any, any> {
+    if (!state.actions.length) {
+      return state;
+    }
+    const _event = initEvent as SCXML.Event<any>;
 
-      return new State<any>({
-        value: stateValue.value,
-        context: stateValue.context,
+    return new State<any>(
+      {
+        value: state.value,
+        context: state.context,
         _event,
         _sessionid: undefined,
-        configuration: stateValue.configuration,
+        configuration: state.configuration,
         transitions: [],
-        children: stateValue.children
-      });
-    }
-
-    return State.from(stateValue, context);
+        children: state.children
+      },
+      state.machine
+    );
   }
 
   /**
    * Creates a new `State` instance that represents the current state of a running machine.
    *
    * @param config
    */
-  constructor(config: StateConfig<TContext, TEvent>) {
-    this.value = config.value;
+  constructor(
+    config: StateConfig<TContext, TEvent>,
+    public machine: AnyStateMachine
+  ) {
     this.context = config.context;
     this._event = config._event;
     this._sessionid = config._sessionid;
+    this._internalQueue = config._internalQueue ?? [];
     this.event = this._event.data;
     this.historyValue = config.historyValue || {};
     this.actions = [...(config.actions || [])];
-    this.meta = getMeta(config.configuration);
     this.matches = this.matches.bind(this);
     this.toStrings = this.toStrings.bind(this);
     this.configuration = config.configuration;
     this.transitions = config.transitions;
     this.children = config.children;
-    this.tags = config.configuration
-      ? getTagsFromConfiguration(config.configuration)
-      : config.tags
-      ? new Set(config.tags)
-      : new Set();
-    this.machine = config.machine;
+
+    this.value = getStateValue(machine.root, config.configuration);
```

Comment:
Value is derived from `configuration` to prevent impossible value/configuration mismatch.